azkaban-web-server/src/main/java/azkaban/scheduler/CustomDeserializerDelegate.java

```diff
@@ -0,0 +1,627 @@
+    return inner.updateFiredTrigger(conn, trigger, state, job);
+  }
+
+  @Override
+  public List<FiredTriggerRecord> selectFiredTriggerRecords(Connection conn, String triggerName, String groupName) throws SQLException {
+    return inner.selectFiredTriggerRecords(conn, triggerName, groupName);
+  }
+
+  @Override
+  public List<FiredTriggerRecord> selectFiredTriggerRecordsByJob(Connection conn, String jobName, String groupName) throws SQLException {
+    return inner.selectFiredTriggerRecordsByJob(conn, jobName, groupName);
+  }
+
+  @Override
+  public List<FiredTriggerRecord> selectInstancesFiredTriggerRecords(Connection conn, String instanceName) throws SQLException {
+    return inner.selectInstancesFiredTriggerRecords(conn, instanceName);
+  }
+
+  @Override
+  public Set<String> selectFiredTriggerInstanceNames(Connection conn) throws SQLException {
+    return inner.selectFiredTriggerInstanceNames(conn);
+  }
+
+  @Override
+  public int deleteFiredTrigger(Connection conn, String entryId) throws SQLException {
+    return inner.deleteFiredTrigger(conn, entryId);
+  }
+
+  @Override
+  public int selectJobExecutionCount(Connection conn, JobKey jobKey) throws SQLException {
+    return inner.selectJobExecutionCount(conn, jobKey);
+  }
+
+  @Override
+  public int insertSchedulerState(Connection conn, String theInstanceId, long checkInTime, long interval) throws SQLException {
+    return inner.insertSchedulerState(conn, theInstanceId, checkInTime, interval);
+  }
+
+  @Override
+  public int deleteSchedulerState(Connection conn, String theInstanceId) throws SQLException {
+    return inner.deleteSchedulerState(conn, theInstanceId);
+  }
+
+  @Override
+  public int updateSchedulerState(Connection conn, String theInstanceId, long checkInTime) throws SQLException {
+    return inner.updateSchedulerState(conn, theInstanceId, checkInTime);
+  }
+
+  @Override
+  public List<SchedulerStateRecord> selectSchedulerStateRecords(Connection conn, String theInstanceId) throws SQLException {
+    return inner.selectSchedulerStateRecords(conn, theInstanceId);
+  }
+
+  @Override
+  public Set<String> selectPausedTriggerGroups(Connection conn) throws SQLException {
+    return inner.selectPausedTriggerGroups(conn);
+  }
+
+  @Override
+  protected Object getObjectFromBlob(ResultSet rs, String colName) throws ClassNotFoundException, IOException, SQLException {
+    byte[] bytes = filterBlob(rs.getBytes(colName));
+    if (bytes != null && bytes.length != 0) {
+      InputStream binaryInput = new ByteArrayInputStream(bytes);
+      ObjectInputStream in = new ObjectInputStream(binaryInput);
+      try {
+        return in.readObject();
+      } finally {
+        in.close();
+      }
+    }
+    return null;
+  }
+
+  private byte[] filterBlob(byte[] original) throws SQLException, IOException {
+    return replace(original,
+        headerBytes(CronSchedule.class, -1330280892166841228L),
```

Comment:
Another nit, why not compute these two once, and store in `private static final  byte[]` fields?

Reply:
No longer applicable; changed workaround approach.